be/src/exprs/vectorized/encryption_functions.cpp

```diff
@@ -199,4 +200,202 @@ ColumnPtr EncryptionFunctions::md5(FunctionContext* ctx, const Columns& columns)
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
+Status EncryptionFunctions::sha2_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    if (scope != FunctionContext::FRAGMENT_LOCAL) {
+        return Status::OK();
+    }
+
+    if (!context->is_constant_column(1)) {
+        return Status::OK();
+    }
+
+    ColumnPtr column = context->get_constant_column(1);
+    auto hash_length = ColumnHelper::get_const_value<TYPE_INT>(column);
+
+    ScalarFunction function;
+    if (hash_length == 224) {
+        function = &EncryptionFunctions::sha224;
+    } else if (hash_length == 256 || hash_length == 0) {
+        function = &EncryptionFunctions::sha256;
+    } else if (hash_length == 384) {
+        function = &EncryptionFunctions::sha384;
+    } else {
```

Comment:
```suggestion
    } else if (hash_length == 512) {
```
Only when hash_length is 512, sha512 will be use.
And for  other hash_length values, you need to implement anther function which generate null values for all inputs.